pkg/summary/summary.go

```diff
@@ -49,119 +58,144 @@ type Summary struct {
-					klog.V(2).Infof("Excluding container %v", containerRecommendation.ContainerName)
-					continue CONTAINER_REC_LOOP
-				}
+			if excludedContainers.Has(containerRecommendation.ContainerName) {
+				klog.V(2).Infof("Excluding container Deployment/%s/%s", dSummary.DeploymentName, containerRecommendation.ContainerName)
+				continue CONTAINER_REC_LOOP
 			}
 
-			var container = containerSummary{
-				ContainerName:  containerRecommendation.ContainerName,
-				UpperBound:     utils.FormatResourceList(containerRecommendation.UpperBound),
-				LowerBound:     utils.FormatResourceList(containerRecommendation.LowerBound),
-				Target:         utils.FormatResourceList(containerRecommendation.Target),
-				UncappedTarget: utils.FormatResourceList(containerRecommendation.UncappedTarget),
-			}
+			var cSummary containerSummary
 			for _, c := range deployment.Spec.Template.Spec.Containers {
 				if c.Name == containerRecommendation.ContainerName {
-					klog.V(6).Infof("Resources for %s: %v", c.Name, c.Resources)
-					container.Limits = utils.FormatResourceList(c.Resources.Limits)
-					container.Requests = utils.FormatResourceList(c.Resources.Requests)
-					break
+					cSummary = containerSummary{
+						ContainerName:  containerRecommendation.ContainerName,
+						UpperBound:     utils.FormatResourceList(containerRecommendation.UpperBound),
+						LowerBound:     utils.FormatResourceList(containerRecommendation.LowerBound),
+						Target:         utils.FormatResourceList(containerRecommendation.Target),
+						UncappedTarget: utils.FormatResourceList(containerRecommendation.UncappedTarget),
+					}
+					cSummary.Limits = utils.FormatResourceList(c.Resources.Limits)
+					cSummary.Requests = utils.FormatResourceList(c.Resources.Requests)
+					klog.V(6).Infof("Resources for Deployment/%s/%s: Requests: %v Limits: %v", dSummary.DeploymentName, c.Name, cSummary.Requests, cSummary.Limits)
 				}
 			}
 
-			deploy.Containers = append(deploy.Containers, container)
+			dSummary.Containers = append(dSummary.Containers, cSummary)
 		}
-		summary.Deployments = append(summary.Deployments, deploy)
+		summary.Deployments = append(summary.Deployments, dSummary)
 	}
 
-	summary.Namespaces = utils.UniqueString(summary.Namespaces)
+	// get the unique list of namespaces we've seen for this summary
+	summary.Namespaces = summaryNamespaces.List()
+
 	return summary, nil
 }
+
+// Update the list of VPAs that the summarizer uses
+func (s *Summarizer) UpdateVPAs() error {
+	nsLog := s.namespace
+	if s.namespace == namespaceAllNamespaces {
+		nsLog = "all namespaces"
+	}
+	klog.V(3).Infof("Looking for VPAs in %s with labels: %v", nsLog, s.vpaLabels)
+	vpas, err := s.listVPAs()
+	if err != nil {
+		klog.Error(err.Error())
+		return err
+	}
+	klog.V(10).Infof("Found vpas: %v", vpas)
+
+	s.vpas = vpas
+	return nil
+}
+
+// Run creates a summary of the vpa info for all namespaces.
+func (s Summarizer) listVPAs() ([]v1beta2.VerticalPodAutoscaler, error) {
```

Comment:
```suggestion
func (s *Summarizer) listVPAs() ([]v1beta2.VerticalPodAutoscaler, error) {
```

Reply:
I'll leave this and the other discussion above up to you, not a blocker.

Reply:
I am actually leaning towards the approach you went with instead. Resolving this. Thanks!